packages/amplication-server/src/core/prismaSchemaParser/prismaSchemaParser.service.ts

```diff
@@ -749,12 +742,88 @@ export class PrismaSchemaParserService {
         (property) => property.type === FIELD_TYPE_NAME
       ) as Field[];
 
+      const hasIdField = modelFields.some(
+        (field) =>
+          field.attributes?.some((attr) => attr.name === ID_ATTRIBUTE_NAME) ??
+          false
+      );
+
+      const hasUniqueField = modelFields.some(
+        (field) =>
+          (field.attributes?.some(
+            (attr) => attr.name === UNIQUE_ATTRIBUTE_NAME
+          ) ??
+            false) &&
+          isValidIdFieldType(field.fieldType as string)
+      );
+
+      // if the model doesn't have any id or unique field that can be used as id filed, we add an id field
+      // The type is the default type for id field in Amplication - String
+      if (!hasIdField && !hasUniqueField) {
+        builder
+          .model(model.name)
+          .field(ID_FIELD_NAME, "String")
+          .attribute(ID_ATTRIBUTE_NAME);
+
+        void actionContext.onEmitUserActionLog(
+          `id field was added to model "${model.name}"`,
+          EnumActionLogLevel.Warning
+        );
+      }
+
+      const uniqueFieldAsIdField = modelFields.find(
+        (field) =>
+          isValidIdFieldType(field.fieldType as string) &&
+          field.attributes?.some((attr) => attr.name === UNIQUE_ATTRIBUTE_NAME)
+      );
```

Comment:
Can this boolean and the one at line 751 be merged into one? I guess we can just use this in both the if statements logic